guides/source/action_controller_overview.md

```diff
@@ -899,6 +899,72 @@ Rails collects all of the parameters sent along with the request in the `params`
 [`query_parameters`]: https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-query_parameters
 [`request_parameters`]: https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-request_parameters
 
+#### `request.variant`
+
+Controllers might need to tailor a response based on context-specific
+information in a request. For example, when a controller responds to a request
+from a mobile platform, it might need to render a variation on the HTML, JSON,
+or XML templates it'd use to respond to a request from a web platform. They can
+incorporate that's necessary to make that decision into the request's
+**variant**.
+
+The request variant is a specialization of the request format, like `:tablet`,
+`:phone`, or `:desktop`.
+
+You can set the [`request.variant`](https://api.rubyonrails.org/classes/ActionDispatch/Http/MimeNegotiation.html#method-i-variant-3D) in a `before_action`:
+
+```ruby
+request.variant = :tablet if request.user_agent.include? "iPad"
+```
+
+Responding to variants in the action is the same as responding to formats:
+
+```ruby
+# app/controllers/projects_controller.rb
+
+def show
+  # ...
+  respond_to do |format|
+    format.html do |html|
+      html.tablet # renders app/views/projects/show.html+tablet.erb
+      html.phone { extra_setup; render ... }
```

Comment:
This might break rubocop because I don't think `render ...` is valid Ruby.

Guides with code-fenced blocks of `ruby` are actually syntax checked and linted, FYI.